util/test/checkChplInstall

```diff
@@ -23,213 +24,146 @@
-    log_error "CHPL_HOME is not a directory: ${CHPL_HOME}"
+elif [ ! -d ${CHPL_HOME} ] ; then
+    err "CHPL_HOME is not a directory: ${CHPL_HOME}"
     exit 1
 else
-    log_debug "CHPL_HOME is set to: ${CHPL_HOME}"
+    echo "Found \$CHPL_HOME directory: ${CHPL_HOME}"
+fi
+
+# Create ~/.chpl directory, if it does not already exist
+CHPL_DIR=${CHPL_CHECK_INSTALL_DIR:-${HOME}/.chpl}
+if [ ! -d ${CHPL_DIR} ] ; then
+    echo "${CHPL_DIR} does not exist. Creating it."
+    mkdir -p ${CHPL_DIR} || { echo "Failed to create ${CHPL_DIR}." && exit 10 ; }
 fi
 
-# Find examples test directory.
-if [ -d $CHPL_HOME/test/release/examples ] ; then
+# Location of test job
+if [ -d ${CHPL_HOME}/test/release/examples ] ; then
     # Install from source repo.
-    TEST_DIR=$CHPL_HOME/test/release/examples
-elif [ -d $CHPL_HOME/examples ] ; then
+    TEST_DIR=${CHPL_HOME}/test/release/examples
+elif [ -d ${CHPL_HOME}/examples ] ; then
     # Install from tarball.
-    TEST_DIR=$CHPL_HOME/examples
+    TEST_DIR=${CHPL_HOME}/examples
 else
-    log_error "Could not find test cases in CHPL_HOME: ${CHPL_HOME}."
+    err "Could not find test cases in CHPL_HOME: ${CHPL_HOME}."
     exit 1
 fi
 
-# For chpldoc, only test the primers/chpldoc.doc.chpl test.
-if [ "${bin_name}" = "chpldoc" ] ; then
-    test_name="chpldoc"
-    TESTS=$TEST_DIR/primers/chpldoc.doc.*
+TMP_TEST_DIR="$(mktemp -d ${CHPL_DIR}/chapel-test-XXXXX)"
+echo "Temporary test job directory: ${TMP_TEST_DIR}"
 
-# List all the files in the examples dir. Do not list any subdirectories,
-# though.
-else
-    test_name="hello world"
-    TESTS=$(find $TEST_DIR -maxdepth 1 -mindepth 1 -type f)
-fi
+TEST_JOB=${TEST_JOB_BASENAME}
 
-# Copy these tests to a temp file.
+# Collect comm protocol environment variables (lowercase to avoid conflicts)
+chpl_comm="$($CHPL_HOME/util/chplenv/chpl_comm.py)"
+chpl_launcher="$($CHPL_HOME/util/chplenv/chpl_launcher.py)"
 
-chpl_dir=${CHPL_CHECK_INSTALL_DIR:-$HOME/.chpl}
-if [ ! -d $chpl_dir ] ; then
-    log_debug "${chpl_dir} does not exist. Creating it."
-    mkdir -p $chpl_dir || { log_error "Failed to create ${chpl_dir}." && exit 10 ; }
-fi
+# Log file for compile output
+TEST_COMP_OUT=${TEST_JOB}.comp.out
 
-TMP_TEST_DIR=$(mktemp -d ${chpl_dir}/chapel-test-XXXXX)
-log_debug "TMP_TEST_DIR: ${TMP_TEST_DIR}"
+# Necessary to suppress warnings when WARNINGS=1 (e.g. nightly build settings)
+COMP_FLAGS="--cc-warnings"
 
-# Ensure the tmp dir is deleted when script exits.
+# Compile test job into temporary directory
+${CHPL_BIN} ${TEST_DIR}/${TEST_JOB}.chpl ${COMP_FLAGS} -o ${TMP_TEST_DIR}/${TEST_JOB} > ${TMP_TEST_DIR}/${TEST_COMP_OUT} 2>&1
+
+# Check that compile was successful
+COMPILE_STATUS=$?
+if [ ${COMPILE_STATUS} -ne 0 ]; then
+    err "Test job failed to compile, Chapel is not installed correctly"
+    echo "Compilation output:"
+    cat ${TMP_TEST_DIR}/${TEST_COMP_OUT}
+    exit 1
+else
+    echo "Test job compiled into ${TMP_TEST_DIR}/${TEST_JOB}"
+fi
```

Comment:
You really should output the contents of TEST_COMP_OUT in case there is a warning in there. I believe the old start_test version would cause a failure if the chpl run printed anything at all to stdout or stderr, but I could be wrong about that.  I'd recommend adding a check for no output from the chpl run, but absolutely print it out if there is any.